verify_test.go

```diff
@@ -25,10 +29,107 @@ func TestVerify(t *testing.T) {
 	if err := p7.Verify(); err != nil {
 		t.Errorf("Verify failed with error: %v", err)
 	}
+
 	expected := []byte("We the People")
 	if !bytes.Equal(p7.Content, expected) {
 		t.Errorf("Signed content does not match.\n\tExpected:%s\n\tActual:%s", expected, p7.Content)
+	}
+}
+
+func TestInvalidSigningTime(t *testing.T) {
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Errorf("failed generating ECDSA key: %v", err)
+	}
+
+	// define certificate validity to a timeframe in the past, so that
+	// the certificate itself is not valid at the time of signing.
+	notBefore := time.Now().UTC().Round(time.Minute).Add(-2 * time.Hour)
+	notAfter := time.Now().UTC().Round(time.Minute).Add(-1 * time.Hour)
+	template := &x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: "TestInvalidSigningtime",
+		},
+		NotBefore:   notBefore,
+		NotAfter:    notAfter,
+		KeyUsage:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageEmailProtection},
+	}
+
+	der, err := x509.CreateCertificate(rand.Reader, template, template, key.Public(), key)
+	if err != nil {
+		t.Errorf("failed creating certificate: %v", err)
```

Comment:
There's a few calls to `t.Errorf` in this code. Maybe you meant `t.Fatalf`?

Reply:
Updated in a few places